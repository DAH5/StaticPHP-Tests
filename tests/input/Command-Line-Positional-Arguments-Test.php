---
friendly_urls: false
---
<?php

$path_to_staticphp_file = "StaticPHP.php";

if( ! is_file( $path_to_staticphp_file ) )
{
    echo "ERROR: path_to_staticphp_file is not a file!";
    return;
}

echo "Beginning Command Line Positional Arguments Test..." . PHP_EOL;

$source_dir_path = "tests/input/_includes/Command-Line-Positional-Arguments-Test-source";
$output_dir_path = "tests/input/_includes/Command-Line-Positional-Arguments-Test-output";
$items_to_ignore = "_ignores";
$friendly_urls = "false";
$metadata_delimiter = "___";
$minify_html = "true";
$minify_css = "true";
$minify_js = "false";
$minify_html_tags_to_preserve = "preserve";
$bulk_redirects_filename = "_bulk_redirections";
$redirections_template_filename = "_redirections_template_file.html";
$minify_css_inplace = "false";
$items_to_passthrough = "_passthrough";

$command = "php StaticPHP.php $source_dir_path $output_dir_path $items_to_ignore $friendly_urls $metadata_delimiter $minify_html $minify_css $minify_js $minify_html_tags_to_preserve $bulk_redirects_filename $redirections_template_filename $minify_css_inplace $items_to_passthrough";


$outputLines = [];
$returnCode = 0;

exec( $command, $outputLines, $returnCode );

echo "Command Executed: " . $command . PHP_EOL;
echo "Exit Code: " . $returnCode . PHP_EOL;
echo "BEGIN OUTPUT" . PHP_EOL . PHP_EOL . implode( PHP_EOL, $outputLines ) . PHP_EOL . PHP_EOL . "END OUTPUT" . PHP_EOL;
echo "...Command Line Positional Arguments Test Complete!" . PHP_EOL;
