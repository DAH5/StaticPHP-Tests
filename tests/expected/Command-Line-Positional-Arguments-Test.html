Beginning Command Line Positional Arguments Test...
Command Executed: php StaticPHP.php tests/input/_includes/Command-Line-Positional-Arguments-Test-source tests/input/_includes/Command-Line-Positional-Arguments-Test-output _ignores false ___ true true false preserve _bulk_redirections _redirections_template_file.html false _passthrough
Exit Code: 0
BEGIN OUTPUT

Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\Command-Line-Positional-Arguments-Test-Example-File.html
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\_passthrough
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\_passthrough\file-will-be-passed-through.html
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\_passthrough
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\friendly-urls
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\friendly-urls\index.html
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\friendly-urls
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\preservation.html
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\one
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\one\index.html
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\one
Emptying Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\two
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\two\index.html
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target\two
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect\target
Done.
Removing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\redirect
Deleting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\stylesheet.css
Done.
Processing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-source
Processing PHP File: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\Command-Line-Positional-Arguments-Test-Example-File.php
Processing MetaData PlaceHolders...
Done.

Processing Functional Blocks...
...Functional Blocks Processed.
Minifying HTML...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\Command-Line-Positional-Arguments-Test-Example-File.html
Processing redirection list...

Minifying HTML...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\/redirect/target/one\index.html
Redirect generated for /redirect/target/one to /redirect/destination/one
Minifying HTML...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\/redirect/target/two\index.html
Redirect generated for /redirect/target/two to /redirect/destination/two

Redirection list processed!

Ignoring Directory Item: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\_ignores
Processing Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\_passthrough
Created New Directory: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\_passthrough
Passing Through File: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\_passthrough\file-will-be-passed-through.html
Done.
Ignoring Directory Item: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\_redirections_template_file.html
Processing HTML File: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\friendly-urls.html
Processing MetaData...

Setting MetaData Key: friendly_urls
with matching value: true

End of MetaData.

Processing MetaData PlaceHolders...
Done.

Processing Functional Blocks...
...Functional Blocks Processed.
Minifying HTML...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\friendly-urls\index.html
Processing HTML File: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\preservation.html
Processing MetaData PlaceHolders...
Done.

Processing Functional Blocks...
...Functional Blocks Processed.
Minifying HTML...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\preservation.html
Minifying CSS File: tests\input\_includes\Command-Line-Positional-Arguments-Test-source\stylesheet.css
Minifying CSS...
Outputting File: tests\input\_includes\Command-Line-Positional-Arguments-Test-output\stylesheet.css
Done.

END OUTPUT
...Command Line Positional Arguments Test Complete!
